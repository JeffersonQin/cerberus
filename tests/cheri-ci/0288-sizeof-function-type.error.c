void foo(void);
int main()
{
  sizeof(foo);
}
