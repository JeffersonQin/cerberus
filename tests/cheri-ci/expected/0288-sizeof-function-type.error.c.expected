cheri-ci/0288-sizeof-function-type.error.c:4:9: error: constraint violation: invalid application of 'sizeof' to a function type
  sizeof(foo);
  ^     ~~~~~ 
ยง6.5.3.4#1, sentence 1: 
1   The sizeof operator shall not be applied to an expression that has function type or an
    incomplete type, to the parenthesized name of such a type, or to an expression that
    designates a bit-field member. The _Alignof operator shall not be applied to a
    function type or an incomplete type.
